specification/loadtestservice/LoadTestService.Management/models.tsp

```diff
@@ -0,0 +1,258 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-providerhub";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ResourceManager;
+using OpenAPI;
+using Autorest;
+using Azure.ResourceManager.Foundations;
+using Azure.ResourceManager.Private;
+
+namespace Microsoft.LoadTestService;
+
+@doc("LoadTest details.")
+model LoadTestResource is TrackedResource<LoadTestProperties> {
+  @doc("Load Test name")
+  @key("loadTestName")
+  @extension("x-ms-parameter-location", "method")
+  @path
+  @segment("loadTests")
+  name: string;
+
+  ...ManagedServiceIdentity;
+}
+
+@doc("Resources provisioning states.")
+enum ResourceState {
+  @doc("Deleted state.")
+  Deleted: "Deleted",
+
+  ...ResourceProvisioningState,
+}
+
+// can be renamed but since the type is already used in published docs, it will be a breaking change.
+@doc("Managed identity type to use for accessing encryption key Url.")
+enum type {
```

Comment:
this is defined in the Azure.ResourceManager.Foundations. Also customer-managed-key and managed idenitity are already in TypeSpec ARM. Please check to see you can align to the common definition.

See [Encryption](https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/reference/data-types#Azure.ResourceManager.Encryption) and [Customer Managed Key](https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/reference/data-types#Azure.ResourceManager.CustomerManagedKeyEncryption)

Reply:
The ManagedIdentityType model in Azure.ResourceManager.Foundations supports 4 values for type whereas we support only 2, therefore we are using the custom model. We couldnt find any other model